### ISSUE TITLE
Deleting a plot removes listener

### ISSUE THREAD

**Issue by ettersi** [MakieOrg/Makie.jl#3542]
Creating a scatter plot from an observable and then deleting the plot deregisters the listeners of the observables:

```julia
julia> x = Observable(Point2f(1,2))
       y = map(identity, x)
       f,a,s = scatter(y)

       println("Before deleting the plot:")
       @show x.listeners
       println()

       delete!(a,s)

       println("After deleting the plot:")
       @show x.listeners;

Before deleting the plot:
x.listeners = Pair{Int64, Any}[0 => Observables.MapCallback(identity, Observable(Float32[1.0, 2.0]), (Observable(Float32[1.0, 2.0]),))]

After deleting the plot:
x.listeners = Pair{Int64, Any}[]
```

I'd say this behaviour is a bug. I don't think plots created from an observable should take ownership of the observable.

**Comment by SimonDanisch** [comment]
That's a bug indeed....
I made a PR to fix this: #3543

**Comment by ettersi** [comment]
Awesome turn around time, thanks a lot! ❤️
